From a7d16c8c84b39f3c4b00b6ea96b0dfc27c05bdfa Mon Sep 17 00:00:00 2001
From: Michael Brown <mbrown@fensystems.co.uk>
Date: Sat, 16 May 2020 19:08:29 +0100
Subject: [PATCH 1/4] rpm: Kill the make_writable function

The SuiteCRM make_writable() function is an abomination from the
depths of hell that brings ten thousand years of shame upon its
author.  Disembowel it gently with a railgun.

Signed-off-by: Michael Brown <mbrown@fensystems.co.uk>
---
 install/install_utils.php | 36 ++----------------------------------
 1 file changed, 2 insertions(+), 34 deletions(-)

diff --git a/install/install_utils.php b/install/install_utils.php
index 6a1f79e90..a5c7e9c72 100755
--- a/install/install_utils.php
+++ b/install/install_utils.php
@@ -1377,40 +1377,8 @@ function insert_default_settings()
 // writable).
 function make_writable($file)
 {
-    $ret_val = false;
-    if (is_file($file) || is_dir($file)) {
-        if (is_writable($file)) {
-            $ret_val = true;
-        } else {
-            $original_fileperms = fileperms($file);
-
-            // add user writable permission
-            $new_fileperms = $original_fileperms | 0x0080;
-            @sugar_chmod($file, $new_fileperms);
-            clearstatcache();
-            if (is_writable($file)) {
-                $ret_val = true;
-            } else {
-                // add group writable permission
-                $new_fileperms = $original_fileperms | 0x0010;
-                @chmod($file, $new_fileperms);
-                clearstatcache();
-                if (is_writable($file)) {
-                    $ret_val = true;
-                } else {
-                    // add world writable permission
-                    $new_fileperms = $original_fileperms | 0x0002;
-                    @chmod($file, $new_fileperms);
-                    clearstatcache();
-                    if (is_writable($file)) {
-                        $ret_val = true;
-                    }
-                }
-            }
-        }
-    }
-
-    return $ret_val;
+    # This code was removed for its own protection
+    return true;
 }
 
 function recursive_make_writable($start_file)
-- 
2.25.1

